### PR TITLE
Fix #1993: Port ju.ConcurrentLinkedQueue from Scala.js

### LINKED ISSUE #1993: Port ConcurrentLinkedQueue from Scala.js
This class is used by [ZIO](https://zio.dev).
It is a blocker for ZIO support on Scala Native.

### PR DESCRIPTION
**Pull request by lolgab**
Fixes: #1993 .
I commented out a test using `Collection#removeIf` because it was failing to link.
Added a comment and a `@Ignore` annotation.
Other than this, I just copy-pasted the files and changed the licensing comment and the package names accordingly to other files in the repository.

I needed to reformat the files. It would be nice if we used the same Scalafmt settings so we could copy-paste the files from Scala.js without edits.

### CODE REVIEW

**Review by sjrd — APPROVED** [review]

**Comment by lolgab** [comment]
This is failing on 2.11 because of the default methods in traits invoked by AbstractCollectionTest etc.
Since there is already a WIP PR for default methods on Scala 2.11 I will wait for it and then rebase this.

**Comment by LeeTibbert** [comment]
Status:  I created separate PRs for CollectionTest, AbstractCollectionTest, & IterableTest and 
             re-ported the code being tested. TrivialImmutableCollectionTest is part of the CollectionTest PR,
             where the former is first used.  This gives a more defined audit trail to my JavaDefaultMethod
             changes.

             I believe those files can be deleted from this PR as soon as the PRs for the moved tests
             are merged.  The PR (#1997) which had originally holding this PR up has been merged.  

            Thank you for giving me permission to  use/move those Tests.  Saved time, effort, & mistakes.
             
              That leaves the  ConcurrentLinkedQueue.scala & ConcurrentLinkedQueueTest.scala.

**Comment by LeeTibbert** [comment]
@lolgab   Now that PR #2013 has been merged, I believe that this PR will build on Scala 2.11
after the conflicting files have been removed & the PR re-based.

Thank you for your understanding & patience while I did some of the JavaDefaultMethod work
and for allowing your port in the PR to be used in the default method PRs.

If there is any difficulty with this PR rebuilding, please let me know.  I expect none, but do
not want you spending time cleaning up after my mistakes.

**Comment by lolgab** [comment]
Wow, the CI is really fast now!

**Comment by lolgab** [comment]
I rebased after your changes without problems @LeeTibbert 
After merging of this one I will check again for other missing parts we need for ZIO support. Hopefully we can get everything we need before the release.

**Comment by lolgab** [comment]
@sjrd Ready for another review round

File: javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala
Comment (sjrd) [inline javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala:23]: In Native we should use a `Long` here instead.

File: javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala
Comment (sjrd) [inline javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala:69]: If `_size` is a Long, this will have to be
```scala
if (_size > Int.MaxValue) Int.MaxValue else _size.toInt
```

**Review by sjrd — CHANGES_REQUESTED** [review]
Just one detail.

**Review by sjrd — APPROVED** [review]

**Pull request merged**
